nucypher/policy/conditions/context.py

```diff
@@ -98,3 +100,17 @@ def get_context_value(context_variable: str, **context) -> Any:
         value = func(**context)  # required inputs here
 
     return value
+
+
+def resolve_any_context_variables(parameters: List[Any], return_value_test, **context):
+    processed_parameters = []
+    for p in parameters:
```

Comment:
Thinking about nested values/lists/tuples. What if the context variable is in a nested list eg.
```python
parameters = [1, 2, [":userAddress", True, "0xdeadbeef"]]
```

(similar for the value for return value test)

Would the context variable be resolved? Basically, I'm wondering if `resolve_any_context_variables` should be recursive? 🤔 

This can be addressed in a separate PR if it proves to be an actual issue, but we should test this case.

Reply:
Yes, it should be addressed recursively, and this PR is a good place for that